Firestore/Source/Local/FSTWriteGroup.mm

```diff
@@ -85,56 +85,60 @@
 
 @interface FSTWriteGroup ()
 - (instancetype)initWithAction:(NSString *)action NS_DESIGNATED_INITIALIZER;
+- (instancetype)initWithAction:(NSString *)action andTransaction:(LevelDBTransaction *)transaction;
 @end
 
 @implementation FSTWriteGroup {
   int _changes;
-  WriteBatch _contents;
 }
 
 + (instancetype)groupWithAction:(NSString *)action {
   return [[FSTWriteGroup alloc] initWithAction:action];
 }
 
++ (instancetype)groupWithAction:(NSString *)action
+                 andTransaction:(firebase::firestore::local::LevelDBTransaction *)transaction {
+  return [[FSTWriteGroup alloc] initWithAction:action andTransaction:transaction];
+}
+
 - (instancetype)initWithAction:(NSString *)action {
   if (self = [super init]) {
     _action = action;
+    _transaction = nullptr;
```

Comment:
This seems like an invalid state now because we're calling things like `_transaction->Delete()` which will crash while this is true.

I think you should remove this initializer.

Reply:
This is still in use by memory persistence.